test/controllers/RouterControllerSpec.scala

```diff
@@ -16,218 +16,126 @@
   val location2 = Location("location2", "/location2")
 
   private val gaToken = "GA-TOKEN"
-  override lazy val fakeApplication = FakeApplication(additionalConfiguration = Map("google-analytics.token" -> gaToken, "location1.path" -> location1.url))
+  override lazy val fakeApplication = FakeApplication(
+    additionalConfiguration = Map("google-analytics.token" -> gaToken, "location1.path" -> location1.url)
+  )
 
-  "router controller" should {
-
-    "return location when location is provided by rules and there is an origin for this location" in new Setup {
-      when(mockThrottlingService.throttle(eqTo(location1), eqTo(auditContext), eqTo(ruleContext))(eqTo(fakeRequest), any[ExecutionContext])).thenReturn(location1)
-      val origin = "some-origin"
-      val controller = new TestRouterController(ruleEngine = ruleEngineStubReturningSomeLocation, throttlingService = mockThrottlingService)
-
-      //when
-      val route = controller.route
-
-      //then
-      val locationWithOrigin = Location(location1.name, location1.url)
-      checkResult(route, locationWithOrigin, "none")
-
-      verify(mockThrottlingService).throttle(eqTo(location1), eqTo(auditContext), eqTo(ruleContext))(eqTo(fakeRequest), any[ExecutionContext])
-    }
+  def withExpectedLogMessages(expectedLogMessages: List[String])(block: => Unit) {
 
-    "return location without origin when location is provided by rules and there is not an origin for this location" in new Setup {
-      when(mockThrottlingService.throttle(eqTo(location1), eqTo(auditContext), eqTo(ruleContext))(eqTo(fakeRequest), any[ExecutionContext])).thenReturn(location1)
+    withCaptureOfLoggingFrom(Logger) { logEvents =>
+      block
 
-      val controller = new TestRouterController(ruleEngine = ruleEngineStubReturningSomeLocation, throttlingService = mockThrottlingService)
-
-      //when
-      val route = controller.route
-
-      //then
-      checkResult(route, location1, "none")
-
-      verify(mockThrottlingService).throttle(eqTo(location1), eqTo(auditContext), eqTo(ruleContext))(eqTo(fakeRequest), any[ExecutionContext])
+      eventually {
+        val logMessages = logEvents.map(_.getMessage)
+        expectedLogMessages.foreach { expectedLogMessage =>
+          logMessages should contain(expectedLogMessage)
+        }
+      }
     }
+  }
 
-    "return default location when location provided by rules is not defined" in new Setup {
-      //given
-      val expectedLocation = BusinessTaxAccount
-
-      //and
-      when(mockThrottlingService.throttle(eqTo(expectedLocation), eqTo(auditContext), eqTo(ruleContext))(eqTo(fakeRequest), any[ExecutionContext])).thenReturn(expectedLocation)
-
-      val controller = new TestRouterController(ruleEngine = ruleEngineStubReturningNoneLocation, throttlingService = mockThrottlingService)
+  "router controller" should {
 
-      //when
-      val route = controller.route
+    "return location" in new Setup {
```

Comment:
the old test file had a few more tests (default location, missing origin) - might be worth checking that none of these are useful